### ISSUE TITLE
Reproducibility issue in computing model logits

### ISSUE THREAD

**Issue by xliu2525** [Profluent-AI/progen3#6]
Hi,

I'm running Progen3-3b model to compute the logits of sequences. I noticed that when I batch sequences differently, the inputs are the same but logits are different, here are code:

```
import torch

from progen3.modeling import ProGen3ForCausalLM
from progen3.batch_preparer import ProGen3BatchPreparer
from progen3.scorer import ProGen3Scorer

model = ProGen3ForCausalLM.from_pretrained("Profluent-Bio/progen3-3b", torch_dtype=torch.bfloat16)
model = model.eval().to("cuda:0")
batch_preparer = ProGen3BatchPreparer()

test_seqs=['AQTI[GLM]0-1-1',
 'AQTI[GLM]1-2-1',
 'AQTI[GLM]2-3-1',
 'AQTI[GLM]3-4-1',
 'RQTI[GLM]0-1-1',
 'RQTI[GLM]1-2-1']

batch_preparer.rng = np.random.default_rng(9) # set seed to choose the same span_id
inputs_4 = batch_preparer.get_batch_kwargs(test_seqs[:4], 
                                           device="cuda:0", reverse=False)
batch_preparer.rng = np.random.default_rng(9) 
inputs_6 = batch_preparer.get_batch_kwargs(test_seqs[:6], 
                                           device="cuda:0", reverse=False)

print(torch.equal(inputs_4["input_ids"], inputs_6["input_ids"][:4]))
print(torch.equal(inputs_4["labels"], inputs_6["labels"][:4]))
print(torch.equal(inputs_4["position_ids"], inputs_6["position_ids"][:4]))
print(torch.equal(inputs_4["sequence_ids"], inputs_6["sequence_ids"][:4]))
```
All above are `True` which means the inputs are the same. Then I run the model:
```
with torch.inference_mode():
    out1 = model(**inputs_4).logits
    out2 = model(**inputs_6).logits

print(torch.allclose(out1[:4], out2[:4]))  # False

diff=out1[:4]-out2[:4]
print(torch.min(diff),torch.max(diff))
```
Min and max of diff is `tensor(-7.1406, device='cuda:0') tensor(2.9717, device='cuda:0')` which means the logits are quite different. Could you help me on this? Thank you so much!

**Comment by sjain-profluent** [comment]
Thanks for bringing this to our attention. We are taking a look at the issue internally and will reach out once we have a mitigation.

**Comment by sjain-profluent** [comment]
Hi, we did some digging and it seems the issue lies with `grouped-gemm` library which we use to optimize our MoE performance. We may update the kernels in the future to potentially mitigate this issue but we don't have a concrete timeline for this right now.
